### PR TITLE
flat-remix-gnome: init at 20210623

### PR DESCRIPTION
**Pull request by VergeDX**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/flat-remix-gnome/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  nativeBuildInputs = [ glib ];
+  installPhase = ''
+    # https://github.com/daniruiz/flat-remix-gnome/blob/20210623/Makefile#L50
+    mkdir -p $out/share/gnome-shell/
+    touch $out/share/gnome-shell/gnome-shell-theme.gresource
+
+    make install PREFIX=$out
+    rm $out/share/gnome-shell/gnome-shell-theme.gresource.old
+  '';
```
Comment (Vonfry) [inline pkgs/data/themes/flat-remix-gnome/default.nix:26]:
```suggestion
  preInstall = ''
    # https://github.com/daniruiz/flat-remix-gnome/blob/20210623/Makefile#L50
    mkdir -p $out/share/gnome-shell/
    touch $out/share/gnome-shell/gnome-shell-theme.gresource
  '';
  
  postInstall = ''
    rm $out/share/gnome-shell/gnome-shell-theme.gresource.old
  '';
```

You can get a detail from [default intall phase in generic builder](https://github.com/NixOS/nixpkgs/blob/6ba632c2a442082f353bf2d7028fda11a888d099/pkgs/stdenv/generic/setup.sh#L1094-L1099)

**Review by Vonfry — COMMENTED** [review]
By the way, commits should be squashed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the contributing guide when naming your commits. This PR should contain one commit for your maintainer addition and one for the package.

**Comment by VergeDX** [comment]
Changes performed. @SuperSandro2000

File: pkgs/data/themes/flat-remix-gnome/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  version = "20210623";
+
+  src = fetchFromGitHub {
+    owner = "daniruiz";
+    repo = pname;
+    rev = version;
+    hash = "sha256-FKsbAvjhXb2ipe3XqACM6OwGYwbBbzvDjsUQYCIQ8NM=";
+  };
+
+  nativeBuildInputs = [ glib ];
```
Comment (jonringer) [inline pkgs/data/themes/flat-remix-gnome/default.nix:18]: doesn't glib need to be in buildInputs if there's compiled schemas?

cc @jtojnar

Comment (jtojnar) [reply pkgs/data/themes/flat-remix-gnome/default.nix:18]: No, `glib-compile-schemas` is just a build time dependency. And so is the setup hook.

File: pkgs/data/themes/flat-remix-gnome/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  makeFlags = [ "PREFIX=$(out)" ];
+  preInstall = ''
+    # https://github.com/daniruiz/flat-remix-gnome/blob/20210623/Makefile#L50
+    mkdir -p $out/share/gnome-shell/
+    touch $out/share/gnome-shell/gnome-shell-theme.gresource
```
Comment (jtojnar) [inline pkgs/data/themes/flat-remix-gnome/default.nix:23]: Why touch it, though?

Comment (VergeDX) [reply pkgs/data/themes/flat-remix-gnome/default.nix:23]: Please check out the code of comment, `make install` step will backup file `gnome-shell-theme.gresource`.
If the file not exists, `make install` will fail. So I touch a fake gresource and delete the backup file.

Comment (jtojnar) [reply pkgs/data/themes/flat-remix-gnome/default.nix:23]: Oh, right. I was reading the wrong branch. An explanation comment might still be useful.

Comment (VergeDX) [reply pkgs/data/themes/flat-remix-gnome/default.nix:24]: Explanation comment added : )

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/128278

1 package built:
flat-remix-gnome
```

**Pull request merged**